cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -248,18 +248,27 @@ impl Environment for ManagedEnvironment {
             .generations()
             .writable(flox.temp_dir.clone())
             .map_err(ManagedEnvironmentError::CreateFloxmetaDir)?;
-        let mut temporary = generations
+
+        let remote = generations
             .get_current_generation()
             .map_err(ManagedEnvironmentError::CreateGenerationFiles)?;
 
+        let mut local_checkout = self.local_env_from_current_generation(flox)?;
```

Comment:
note nonblocking: I found the naming of `local_env_from_current_generation` and the comments really confusing - I think `local_env_or_current_generation` would make more sense, and
`Create a local checkout of the current generation` should be changed to `Return the local environment or create a local checkout of the current generation`